### ISSUE TITLE
BE core dump when ShardedLRUCache::prune 

### ISSUE THREAD

**Issue by kangkaisen** [apache/incubator-doris#1123]
**Describe the bug**

```
./start_be.sh: line 58: kill: (33824) - No such process
start time: Mon Apr 29 16:46:45 CST 2019
*** Aborted at 1557391870 (unix time) try "date -d @1557391870" if you are using GNU date ***
PC: @           0xd2cdf7 doris::ShardedLRUCache::prune()
*** SIGSEGV (@0x0) received by PID 38777 (TID 0x7f9cd32e4700) from PID 0; stack trace: ***
    @     0x7f9cde7632f0 (unknown)
    @           0xd2cdf7 doris::ShardedLRUCache::prune()
    @           0xc5b75e doris::OLAPEngine::start_clean_fd_cache()
    @           0xc82caf doris::OLAPEngine::_fd_cache_clean_callback()
    @           0xc82d5f _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJZN5doris10OLAPEngine16_start_bg_workerEvEUlvE4_EEEEE6_M_runEv
    @          0x2526acf execute_native_thread_routine
    @     0x7f9cde518e25 start_thread
    @     0x7f9cde82bbad __clone



(gdb) bt full
#0  0x0000000000d2cdf7 in _lru_remove (e=<optimized out>, this=0x8b143c8) at /home/kangkaisen/palo/be/src/olap/lru_cache.cpp:224
No locals.
#1  _finish_erase (e=<optimized out>, this=0x8b143c8) at /home/kangkaisen/palo/be/src/olap/lru_cache.cpp:295
No locals.
#2  prune (this=0x8b143c8) at /home/kangkaisen/palo/be/src/olap/lru_cache.cpp:314
        e = 0x2b63e24d0
        num_prune = 206
#3  doris::ShardedLRUCache::prune (this=0x8b14000) at /home/kangkaisen/palo/be/src/olap/lru_cache.cpp:376
        s = <optimized out>
        num_prune = 2748
#4  0x0000000000c5b75e in doris::OLAPEngine::start_clean_fd_cache (this=this@entry=0x5397c00)
    at /home/kangkaisen/palo/be/src/olap/olap_engine.cpp:1691
No locals.
#5  0x0000000000c82caf in doris::OLAPEngine::_fd_cache_clean_callback (this=0x5397c00, arg=arg@entry=0x0)
    at /home/kangkaisen/palo/be/src/olap/olap_server.cpp:101
        interval = 3600
```

**Comment by imay** [comment]
Did this happen when you killed process? or when it runs normally?

**Comment by kangkaisen** [comment]
> Did this happen when you killed process? or when it runs normally?

it runs normally

**Comment by kangpinghuang** [comment]
hi, which branch or tag of code you use?

**Comment by kangkaisen** [comment]
> hi, which branch or tag of code you use?

The base commit is 
```
commit ae8d16c81e47930ae0f1fb0c31d35dcfe9f337dc
Author: chenhao7253886 <510341142@qq.com>
Date:   Mon Nov 12 11:15:39 2018 +0800

    Fix failed cases in regression test (#299)
```
